src/modules/test/test.reducer.ts

```diff
@@ -0,0 +1,61 @@
+import * as preTestDeclarationActions from './pre-test-declarations/pre-test-declarations.actions';
+import * as testOutcomeActions from '../../pages/journal/components/test-outcome/test-outcome.actions';
+import {
+  initialState as preTestDeclarationsInitialState,
+  preTestDeclarationsReducer,
+} from './pre-test-declarations/pre-test-declarations.reducer';
+import { candidateReducer, initialState as candidateInitialState } from './candidate/candidate.reducer';
+import { get } from 'lodash';
+
+// Extend this with any new test domain action types
+type TestAction = preTestDeclarationActions.Types | testOutcomeActions.Types;
+
+const initialState = {};
+
+export const testReducer = (
+  state = initialState,
+  action: TestAction,
+) => {
+  const slotId = deriveSlotId(state, action);
+  if (!slotId) {
+    return state;
+  }
+
+  const existingStateForSlot = state[slotId];
+  const applyReducer = makeReducerApplyer(existingStateForSlot, state, slotId, action);
+
+  return {
+    ...state,
+    [slotId]: {
+      preTestDeclarations: applyReducer(
+        preTestDeclarationsReducer,
+        'preTestDeclarations',
+        preTestDeclarationsInitialState,
+      ),
+      candidate: applyReducer(
+        candidateReducer,
+        'candidate',
+        candidateInitialState,
+      ),
+    },
+    current: {
+      slotId,
+    },
+  };
+};
+
+const deriveSlotId = (state, action): string | null => {
+  if (action instanceof testOutcomeActions.TestOutcomeStartTest) {
```

Comment:
How come we've departed from the standard `switch` statement pattern for reacting to action types??

Reply:
I think the reason for this departure is that the test reducer is serving a different purpose. The rest of our reducers are "leaf" reducers and are taking actions and directly manipulating one piece of state. This is a parent reducer which delegates most of its manipulation to child reducers. I didn't see much need to `switch`, as all actions are passed to every listed reducer and _the children_ decide whether they care about that action by switching themselves. Perhaps there's a way we could refactor to make it more consistent with our other reducers?

Reply:
Makes perfect sense, this is a common concept in redux. Have a look at this page https://redux.js.org/recipes/structuring-reducers/splitting-reducer-logic, there's a built-in function `combineReducers` that does what you're looking for I think, i.e. splitting up reducers and having them operate on slices of nested state, looks to be available in `ngrx` also - https://ngrx.io/api/store/combineReducers

Reply:
Managed to simplify the code by using `combineReducers` for the individual test scoped state, but still need the custom wrapper around that to navigate to the correct test.